src/scripts/show_originals.js

```diff
@@ -6,36 +6,58 @@ import { keyToCss } from '../util/css_map.js';
 
 const excludeClass = 'xkit-show-originals-done';
 const hiddenClass = 'xkit-show-originals-hidden';
+const activeTimelineClass = 'xkit-show-originals-timeline';
 const lengthenedClass = 'xkit-show-originals-lengthened';
 
-const dashboardTimeline = '/v2/timeline/dashboard';
-const dashboardTimelineRegExp = new RegExp(dashboardTimeline);
-
-const lengthenTimelines = async () => {
-  const timeline =
-    document.querySelector(`[data-timeline="${dashboardTimeline}"]:not(.${excludeClass})`);
+let showOwnReblogs;
+let showReblogsWithContributedContent;
+let whitelistedUsernames;
+let runOnDashboard;
+let runOnPeepr;
+let runOnBlogSubscriptions;
 
-  if (timeline) {
-    timeline.classList.add(excludeClass);
-    const paginationCss = await keyToCss('manualPaginatorButtons');
+const lengthenTimeline = async (timeline) => {
+  const paginatorSelector = await keyToCss('manualPaginatorButtons');
 
-    if (!timeline.querySelector(paginationCss)) {
-      timeline.classList.add(lengthenedClass);
-    }
+  if (!timeline.querySelector(paginatorSelector)) {
+    timeline.classList.add(lengthenedClass);
   }
 };
 
-let showOwnReblogs;
-let showReblogsWithContributedContent;
-let whitelistedUsernames;
+const processTimelines = () => {
+  [...document.querySelectorAll(`[data-timeline]:not(.${excludeClass})`)]
+    .forEach(timeline => {
+      timeline.classList.add(excludeClass);
+
+      const onDashboard = timeline.dataset.timeline === '/v2/timeline/dashboard';
+      const isSinglePostPeepr = timeline.dataset.timeline.includes('permalink');
+      const onPeepr = timeline.closest('[role="dialog"]') !== null && isSinglePostPeepr === false;
+      const onBlogSubscriptions =
+        timeline.dataset.timeline === '/v2/timeline' &&
+        timeline.dataset.which === 'blog_subscriptions';
+
+      const shouldRun =
+        (runOnDashboard && onDashboard) ||
+        (runOnPeepr && onPeepr) ||
+        (runOnBlogSubscriptions && onBlogSubscriptions);
+
+      if (shouldRun) {
+        timeline.classList.add(activeTimelineClass);
+        lengthenTimeline(timeline);
+      }
+    });
+};
 
 const processPosts = async function () {
   const whitelist = whitelistedUsernames.split(',').map(username => username.trim());
 
   await exposeTimelines();
-  lengthenTimelines();
+  processTimelines();
+
+  const postElements = getPostElements({ excludeClass, includeFiltered: true })
+    .filter(postElement => postElement.closest('[data-timeline]').classList.contains(activeTimelineClass));
```

Comment:
Would there be a performance draw from changing this to

```suggestion
    .filter(postElement => postElement.matches(`[data-timeline].${activeTimelineClass} div');
```

?

Reply:
Shouldn't make any difference :) Anything that gets run like 10 times can be done however; it's the querySelectors that get run in some fashion on like 30,000 DOM nodes where performance becomes at all significant!

Reply:
Oh, the other thought I had that's sort of related to this is that, in theory, you could skip this check entirely, run the extension on every timeline, and just change the selector to `${activeClass} ${hiddenClass} article { display: none; }` or whatever. Can't really think of a reason to do it that way, but it's probably a possibility for all of the exposeTimelines scripts.